include/ginkgo/core/base/types.hpp

```diff
@@ -51,30 +51,8 @@
 #endif
 
 
-#if (defined(__CUDA_ARCH__) && defined(__APPLE__)) || \
-    defined(__HIP_DEVICE_COMPILE__)
-
-#ifdef NDEBUG
-#define GKO_ASSERT(condition) ((void)0)
-#else  // NDEBUG
-// Poor man's assertions on GPUs for MACs. They won't terminate the program
-// but will at least print something on the screen
-#define GKO_ASSERT(condition)                                               \
-    ((condition)                                                            \
-         ? ((void)0)                                                        \
-         : ((void)printf("%s: %d: %s: Assertion `" #condition "' failed\n", \
-                         __FILE__, __LINE__, __func__)))
```

Comment:
Was this giving issues because of some extra `%` in the `condition`?

Reply:
No, it used to be an issue in the batch kernels, but I refactored that to fix the warning. But the ROCm assertions were broken for a long time, now the workaround is causing issues